src/services/ethereum/processor.ts

```diff
@@ -132,4 +143,16 @@ export class Processor {
 
     return false;
   }
+
+  private async markLowerPriorityTransactions(scheduled: IScheduled, transactonList: IScheduled[]) {
+    for (const transaction of transactonList) {
+      if (
+        transaction._id !== scheduled._id &&
+        transaction.nonce === scheduled.nonce &&
+        transaction.priority > scheduled.priority
+      ) {
+        transaction.status = Status.StaleNonce;
```

Comment:
You need to persist the changes in the DB. See e.g. line 138 for an example - `scheduled.update({ ... })`